checkov/kubernetes/graph_builder/graph_components/edge_builders/LabelSelectorEdgeBuilder.py

```diff
@@ -1,16 +1,17 @@
 from __future__ import annotations
 
-from checkov.kubernetes.graph_builder.graph_components.K8SEdgeBuilder import K8SEdgeBuilder
+from checkov.kubernetes.graph_builder.graph_components.edge_builders.K8SEdgeBuilder import K8SEdgeBuilder
 from checkov.kubernetes.graph_builder.graph_components.blocks import KubernetesBlock
+from checkov.kubernetes.kubernetes_utils import FILTERED_RESOURCES_FOR_EDGE_BUILDERS
 
 
 class LabelSelectorEdgeBuilder(K8SEdgeBuilder):
 
     @staticmethod
     def should_search_for_edges(vertex: KubernetesBlock) -> bool:
-        if vertex.metadata is not None and vertex.metadata.labels is not None:
-            return True
-        return False
+        return vertex.metadata is not None \
+            and vertex.metadata.labels is not None \
+            and vertex.attributes.get("kind") not in FILTERED_RESOURCES_FOR_EDGE_BUILDERS
```

Comment:
does it make sense to search for edges if we don't have `kind`? if we assume it is always there, then directly access it `vertex.attributes["kind"]`

Reply:
unfortunately not all object have this attribute, hence [this](https://github.com/bridgecrewio/checkov/pull/3645)

Reply:
ok, then wouldn't it be better to do this
```suggestion
            and "kind" in vertex.attributes
            and vertex.attributes["kind"] not in FILTERED_RESOURCES_FOR_EDGE_BUILDERS
```
just want to make sure, if we need to consider nodes, which have no `kind` attribute